events_test/src/test/java/com/adobe/aio/event/management/RegistrationServiceIntegrationTest.java

```diff
@@ -51,24 +52,19 @@ public void createGetDeleteJournalRegistration() {
     try {
       providerId = providerServiceTester.createOrUpdateProvider(TEST_EVENT_PROVIDER_LABEL,
           TEST_EVENT_CODE).getId();
-      Registration registration = createJournalRegistration(TEST_REGISTRATION_NAME, providerId,
+      Registration registration = createOrUpdateJournalRegistration(TEST_REGISTRATION_NAME, providerId,
           TEST_EVENT_CODE);
       registrationId = registration.getRegistrationId();
-      Optional<Registration> found = registrationService.findById(registrationId);
-      assertTrue(found.isPresent());
-      logger.info("Found AIO Event Registration: {}", found.get());
-      assertEquals(registrationId, found.get().getRegistrationId());
-      assertEquals(registration.getClientId(), found.get().getClientId());
-      assertEquals(registration.getDescription(), found.get().getDescription());
-      assertEquals(registration.getName(), found.get().getName());
-      assertEquals(registration.getDeliveryType(), found.get().getDeliveryType());
-      assertEquals(registration.getEventsOfInterests(),
-          found.get().getEventsOfInterests());
-      assertEquals(registration.getWebhookStatus(), found.get().getWebhookStatus());
-      assertEquals(registration.isEnabled(), found.get().isEnabled());
-      assertEquals(registration.getWebhookUrl(), found.get().getWebhookUrl());
-      assertEquals(registration.getJournalUrl().getHref(), found.get().getJournalUrl().getHref());
-      assertEquals(registration.getTraceUrl().getHref(), found.get().getTraceUrl().getHref());
+      String finalRegistrationId = registration.getRegistrationId();
+      assertCreatedOrUpdatedRegistrationMatchesWithFoundRegistration(registration,
+          () -> registrationService.findById(finalRegistrationId));
+
+      // covering the update path
+      registration = createOrUpdateJournalRegistration(TEST_REGISTRATION_NAME, providerId,
```

Comment:
could you add the same coverage for the `ProviderService` pretty please ?

Reply:
Well,I have done this [here](https://github.com/adobe/aio-lib-java/pull/200/commits/8f19718e7ef1057a50dda92a76e061fb4fc17bff#r1450096984) in the `ProviderServiceIntegrationTest`, is there any other place I need to do that you meant?

Reply:
oh I did not catch this, thanks !